includes/class-wc-subscriptions-tracker.php

```diff
@@ -170,28 +170,32 @@ private static function get_subscription_orders() {
 	/**
 	 * Gets first and last subscription created dates.
 	 *
-	 * @return array
+	 * @return array 'first' and 'last' created subscription dates as a string in the date format 'Y-m-d H:i:s' or '-'.
 	 */
 	private static function get_subscription_dates() {
-		global $wpdb;
-
-		$min_max = $wpdb->get_row(
-			"
-			SELECT
-				MIN( post_date_gmt ) as 'first', MAX( post_date_gmt ) as 'last'
-			FROM {$wpdb->prefix}posts
-			WHERE post_type = 'shop_subscription'
-			AND post_status NOT IN ( 'trash', 'auto-draft' )
-		",
-			ARRAY_A
+		// Ignore subscriptions with status 'trash'.
+		$first = wcs_get_subscriptions(
```

Comment:
We're replacing a single SQL query with two queries using `wcs_get_subscriptions()`.

I'm unsure if this will have a noticeable impact on execution speed.

Is there a way to compress this into a single query using HPOS-compatible functions?

Reply:
This is calculated [once per week by default](https://github.com/woocommerce/woocommerce/blob/2de5a8f08657a70987f48421dc43706890b96e4d/plugins/woocommerce/includes/class-wc-tracker.php#L48-L60), so it may not be a problem to run one extra query.

Reply:
@Jinksi I think these two queries are fine.

If we're concerned about performance, one option would be to do something similar to what we did in the `wcs_get_subscriptions_for_product()` function to support the direct DB queries in both CPT and HPOS environments.

For this particular query, we'd need to set variables `$date_column`, `$orders_tables_name`, `$order_type_column` and `$order_status_column` depending on whether HPOS is enabled.

I don't think there's a big concern around running these two queries versus one direct one. I kinda like using functions so that they're more future proof (hopefully 😄)